syntax = "proto3";
package streetlevel;

message LatLng {
  double lat = 1;
  double lng = 2;
}

enum MapsResultType {
  MAPS_RESULT_TYPE_UNKNOWN = 0;
  MAPS_RESULT_TYPE_PLACE = 1;
  MAPS_RESULT_TYPE_COLLECTION = 2;
  MAPS_RESULT_TYPE_PUBLISHER = 3;
  MAPS_RESULT_TYPE_PLACE_QUESTIONNAIRE = 4;
  MAPS_RESULT_TYPE_BATCH_REVERSE_GEOCODE = 5;
}

enum RequestType {
    REQUEST_TYPE_UNKNOWN = 0;
    REQUEST_TYPE_SEARCH = 1;
    REQUEST_TYPE_GEOCODING = 2;
    REQUEST_TYPE_CANONICAL_LOCATION_SEARCH = 3;
    REQUEST_TYPE_REVERSE_GEOCODING = 4;
    REQUEST_TYPE_PLACE_LOOKUP = 5;
    REQUEST_TYPE_MERCHANT_LOOKUP = 6;
    REQUEST_TYPE_PLACE_REFINEMENT = 7;
    REQUEST_TYPE_SIRI_SEARCH = 8;
    REQUEST_TYPE_LOCATION_DIRECTED_SEARCH = 9;
    REQUEST_TYPE_AUTOCOMPLETE = 10;
    REQUEST_TYPE_BROWSE_CATEGORY_SUGGESTION = 11;
    REQUEST_TYPE_CATEGORY_SEARCH = 12;
    REQUEST_TYPE_POPULAR_NEARBY = 13;
    REQUEST_TYPE_ZERO_KEYWORD_CATEGORY_SUGGESTION = 14;
    REQUEST_TYPE_SEARCH_FIELD_PLACEHOLDER = 15;
    REQUEST_TYPE_BATCH_POPULAR_NEARBY_SEARCH = 16;
    REQUEST_TYPE_VENDOR_SPECIFIC_PLACE_REFINEMENT = 17;
    REQUEST_TYPE_NEARBY_SEARCH = 18;
    REQUEST_TYPE_ADDRESS_OBJECT_GEOCODING = 19;
    REQUEST_TYPE_ZERO_KEYWORD_WITH_SEARCH_RESULTS_SUGGESTION = 20;
    REQUEST_TYPE_EXTERNAL_TRANSIT_LOOKUP = 21;
    REQUEST_TYPE_FEATURE_ID_GEOCODING = 22;
    REQUEST_TYPE_MAPS_IDENTIFIER_PLACE_LOOKUP = 23;
    REQUEST_TYPE_DATASET_STATUS_CHECK = 24;
    REQUEST_TYPE_OFFLINE_AREA_LOOKUP = 25;
    REQUEST_TYPE_BATCH_REVERSE_GEOCODING = 26;
    REQUEST_TYPE_OFFLINE_SUGGESTED_REGIONS = 27;
    REQUEST_TYPE_OFFLINE_POLYGON_QUAD_NODES = 28;
    REQUEST_TYPE_OFFLINE_PROACTIVE_REGIONS = 29;
    REQUEST_TYPE_BRAND_LOOKUP = 30;
    REQUEST_TYPE_OFFLINE_UPDATE_MANIFEST = 31;
    REQUEST_TYPE_WIFI_FINGERPRINT_LOOKUP = 32;
    REQUEST_TYPE_INITIAL_OFFLINE_SUGGESTION = 33;
    REQUEST_TYPE_IP_GEO_LOOKUP = 34;
    REQUEST_TYPE_GROUND_VIEW_LABEL= 35;
    REQUEST_TYPE_BATCH_SPATIAL_LOOKUP = 36;
    REQUEST_TYPE_TRANSIT_VEHICLE_POSITION = 37;
    REQUEST_TYPE_PLACE_COLLECTION_LOOKUP = 38;
    REQUEST_TYPE_TRANSIT_SCHEDULE_LOOKUP = 39;
    REQUEST_TYPE_BATCH_CATEGORY_LOOKUP = 40;
    REQUEST_TYPE_BATCH_MERCHANT_LOOKUP_BRAND = 41;
    REQUEST_TYPE_CHILD_PLACE_LOOKUP_BY_CATEGORY = 42;
    REQUEST_TYPE_COLLECTION_SUGGESTION = 43;
    REQUEST_TYPE_MAPS_SEARCH_HOME = 44;
    REQUEST_TYPE_PLACE_QUESTIONNAIRE_LOOKUP = 45;
    REQUEST_TYPE_PUBLISHER_VIEW = 46;
    REQUEST_TYPE_ALL_COLLECTIONS_VIEW = 47;
    REQUEST_TYPE_AIRPORT_ENTITY_PLACE_LOOKUP = 48;
    REQUEST_TYPE_TERRITORY_LOOKUP = 49;
    REQUEST_TYPE_TRANSIT_NEARBY_SCHEDULE_LOOKUP = 50;
    REQUEST_TYPE_MAPS_HOME = 51;
    REQUEST_TYPE_ALL_GUIDES_LOCATIONS = 52;
    REQUEST_TYPE_GUIDES_HOME = 53;
    REQUEST_TYPE_EXTENDED_GEO_LOOKUP = 54;
    REQUEST_TYPE_QUERY_UNDERSTANDING = 55;
    REQUEST_TYPE_POI_AT_ADDRESS_LOOKUP = 56;
    REQUEST_TYPE_TRANSIT_NEARBY_PAYMENT_METHOD_LOOKUP = 57;
    REQUEST_TYPE_PLACECARD_ENRICHMENT = 58;
}
